### ISSUE TITLE
Support refseqs stored as faidx-indexed fasta files

### ISSUE THREAD

**Issue by benwbooth** [GMOD/jbrowse#317]
samtools has a faidx option that can produce an index for a fasta file, as documented here:

http://samtools.sourceforge.net/samtools.shtml

It would be really cool if JBrowse could read refseqs in plain fasta format using these index files. That would make it unnecessary to run the prepare-refseqs.pl script, and you could just have a single fasta file and a .fai index file instead of a bunch of directories and files to store your refseqs.

**Comment by keiranmraine** [comment]
Should this still be open?

Indexing of normal fasta is now supported, but not the compressed version.  The newer version of samtools expects these to be compressed with bgzip instead of razf.

**Comment by cmdcolin** [comment]
This could probably be closed. Technically bgzip of the reference fasta wasn't added with the current indexed fasta but it works well after #495 was merged

**Comment by keiranmraine** [comment]
@cmdcolin , some confusion here.  Are you expecting bgzip of the reference using the very old technique from legacy samtools with an `*.fa.gz.fai` file or using the new bgzip from htslib `fa.gz.gzi`?

The new technique expects a 'gzi' file for interoperability with `samtools faidx` and that is not compatible with `prepare-refseqs.pl`:

```
$ bin/prepare-refseqs.pl --seqType dna --out wibble/Homo_sapiens/GRCh37 --indexed_fasta wibble/tmp/Homo_sapiens/GRCh37/85/genome.fa.gz
Unable to read from wibble/tmp/Homo_sapiens/GRCh37/85/genome.fa.gz.fai: No such file or directory
```

when renamed to fai:

```
$ bin/prepare-refseqs.pl --seqType dna --out wibble/Homo_sapiens/GRCh37 --indexed_fasta wibble/tmp/Homo_sapiens/GRCh37/85/genome.fa.gz
Improperly-formatted line in fai file (../Homo_sapiens/GRCh37/85/genome.fa.gz.fai):
????????
```

If compressed with htslib bgzip and indexed with samtools faidx `prepare-refseqs.pl` completes but the `*.fa.gz` is not recognised by the interface.

**Comment by keiranmraine** [comment]
FYI the method of generating a bgzip fasta is as follows:

```
$ bgzip genome.fa
$ samtools faidx genome.fa.gz
```

Results in files:

```
$ ls -ltrh
total 843M
-rw-r----- 1 kr2 cancer 843M Aug  9 15:57 genome.fa.gz
-rw-r----- 1 kr2 cancer 755K Aug 10 10:27 genome.fa.gz.gzi
-rw-r----- 1 kr2 cancer 2.7K Aug 10 10:27 genome.fa.gz.fai
```

If this isn;t supported it's fine, I just need to know if it was intended to be as I'll need to request far more storage on the server if not.

**Comment by cmdcolin** [comment]
@keiranmraine actually neither. the only thing that is implemented to my knowledge is uncompressed indexed fasta, so basically just a regular fasta file that has been indexed, e.g. samtools faidx regular_fasta.fa

Note that the twobit pull request https://github.com/GMOD/jbrowse/pull/759 provides optimal compression and could be tested too. It works great in my experience with it

**Comment by keiranmraine** [comment]
Yup, looks like 2bit would be as good an option for me.

**Comment by cmdcolin** [comment]
Faidx supported since 1.12.0 and now twobit can be added in $nextversion.

Bgzipped+fasta index will probably be separate issue
